### DISCUSSION TITLE
how to access google bucket (gcr.io or gs)

### DISCUSSION THREAD

**Discussion by moonsyim** [argoproj/argo-cd#7653] [Q&A]
Hello,  would you please help me to access google bucket as helm repository.
Trying to deploy kafka with zookeeper,    zookeeper is by dependency with requirement.yaml

# argocd version
            argocd: v2.1.3+d855831

# kubectl get configmap -n argocd
        NAME                                  DATA   AGE
        argocd-cm                          1      26d
        argocd-cmd-params-cm         0      26d
        argocd-gpg-keys-cm              0      26d
        argocd-rbac-cm                     0      26d
        argocd-ssh-known-hosts-cm   1      26d
        argocd-tls-certs-cm                0      26d
        kube-root-ca.crt                     1      26d

Reference is   https://argo-cd-docs.readthedocs.io/en/latest/operator-manual/declarative-setup/
                    https://argocd-image-updater.readthedocs.io/en/stable/configuration/registries/

test 1)  configmap  argocd-cm    -------------------------------------------
apiVersion: v1
data:
  registries.conf: |
    registries:
    - name: Google Container Registry
      api_url: https://gcr.io
      prefix: gcr.io
      ping: no
      credentials: imagePullSecrets: "secret is here"

requriement.yaml

dependencies:
  - name: zookeeper
    repository: gcr.io://"my bucket path here"
    version: 3.4.13    
    tags:
      - zookeeper

Result is
   Unknown desc = `helm dependency build` failed exit status 1: Error: could not find protocol handler for: gcr.io
   

.
.

test 2)    configmap  argocd-cm   -------------------------------------------
apiVersion: v1
data:
  registries.conf: |
    registries:
    - name: Google Container Registry
      api_url: https://storage.googleapis.com/"my bucket path here"
      prefix: helm-argo
      ping: no
      credentials: imagePullSecrets:"secret is here"

requriement.yaml

dependencies:
  - name: zookeeper
    repository: helm-repo
    version: 3.4.13    
    tags:
      - zookeeper

Result is
    no repository definition for helm-argo. Please add them via 'helm repo add'

.
.

test 3)  configmap   argocd-cm    ----------------------------------------------------

apiVersion: v1
data:
  repositories: |
    - type: helm
      url: https://storage.googleapis.com/"my bucket path here"
      name: helm-argo
      credentials: imagePullSecrets:"secret is here"

requirement.yaml

dependencies:
  - name: zookeeper
    repository: helm-repo
    version: 3.4.13    
    tags:
      - zookeeper

Result is
is not a valid chart repository or cannot be reached: failed to fetch https://storage.googleapis.com/"my bucket path"/index.yaml : 403 Forbidden

.
.

test 4)      adding repo -------------------------------------------

argocd repo add  https://storage.googleapis.com/"my bucket path" --type helm --name  helm-argo
     FATA[0000] rpc error: code = Unknown desc = failed to get index: 403 Forbidden

**Comment by tyriis** [comment]
You can patch the docker images and install a plugin for gcs backed helm repo https://github.com/hayorov/helm-gcs
